modules/virt-creating-a-service-backed-by-virtual-machines.adoc

```diff
@@ -0,0 +1,142 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/vm_networking/virt-using-the-default-pod-network-with-virt.adoc
+
+[id="virt-creating-a-service-backed-by-virtual-machines_{context}"]
+
+= Creating a service backed by virtual machines
+
+To create a service backed by a virtual machine, you start the `VirtualMachineInstance`,
+if it is not already running. Once the instance is running, you connect to the instance, and
+create a service object to expose the instance.
+
+The following service types are supported:
+
+* `ClusterIP` - Connect using service IP and service port inside the cluster network.
+* `NodePort` - Connect using node IP and node port outside the cluster network.
+* `LoadBalancer` - Connect using public IP and port.
+
+This procedure presents an example of how to create and expose a service object of type `ClusterIP`
+as a virtual machine-backed service.
+
+[NOTE]
+====
+Labels on a `VirtualMachineInstance` are passed through to the pod. The labels on
```

Comment:
Would it make sense to mention explicitly the virt-launcher pod?

Reply:
@rhrazdil - If so, please supple information on what the virt-launcher is and how it works. Is it documented somewhere upstream?

Reply:
I don't think we should mention our implementation details here. For the user, the only important part is that the mapping is between service selector and VM labels. Explaining how it happens on the background may only confuse them.

Reply:
It sounds like @phoracek says no on this.